core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -163,16 +163,16 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
     }
 
     /*package*/ void removeComputer(final Computer computer) {
-        Queue.withLock(new Runnable() {
-            @Override
-            public void run() {
-                Map<Node,Computer> computers = getComputerMap();
-                for (Map.Entry<Node, Computer> e : computers.entrySet()) {
-                    if (e.getValue() == computer) {
-                        computers.remove(e.getKey());
-                        computer.onRemoved();
-                        return;
+        Queue.withLock(() -> {
+            Map<Node,Computer> computers = getComputerMap();
+            for (Map.Entry<Node, Computer> e : computers.entrySet()) {
+                if (e.getValue() == computer) {
+                    computers.remove(e.getKey());
```

Comment:
Should it happen after the listeners call ?

Reply:
`computer.onRemoved();` is called after, so listener also

Reply:
It doesn't seem congruent with the Javadoc for onRemoved()?